src/test/java/com/epam/jamp2/service/impl/CalculationCommandExecutionServiceIntegrationTest.java

```diff
@@ -0,0 +1,92 @@
+package com.epam.jamp2.service.impl;
+
+import com.epam.jamp2.model.CalculationCommand;
+import com.epam.jamp2.model.Operation;
+import com.epam.jamp2.model.UnknownCurrencyException;
+import com.epam.jamp2.model.Value;
+import com.epam.jamp2.rest.FixerioServiceProxy;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThan;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@EnableConfigurationProperties
+
+public class CalculationCommandExecutionServiceIntegrationTest {
+
+    @Autowired
+    private CalculationCommandExecutionServiceImpl calculationCommandExecutionServiceImpl;
+
+    @Autowired
+    FixerioServiceProxy fixerioServiceProxy;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    /**
+     * Test case to generate the real rates data for unit test cases
+     *
+     * @throws IOException write rates data to disk failed
+     */
+    @Test
+    public void testRestApiResultsAndSetupForUnitTest() throws IOException {
```

Comment:
I got your point about automated test data preparation here, but what I don't like about it is I can't see that test data. So I would recommend populating rates right inside your tests (I believe that specifying rates for 3 currencies is pretty enough for all the cases) or provide readable file. It will simplify your code a lot and also increase readability and easiness to understand tests code.